processor/resourcedetectionprocessor/internal/kubeadm/kubeadm.go

```diff
@@ -0,0 +1,61 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package kubeadm // import "github.com/open-telemetry/opentelemetry-collector-contrib/processor/resourcedetectionprocessor/internal/kubeadm"
+
+import (
+	"context"
+	"fmt"
+
+	"go.opentelemetry.io/collector/pdata/pcommon"
+	"go.opentelemetry.io/collector/processor"
+	conventions "go.opentelemetry.io/collector/semconv/v1.6.1"
+	"go.uber.org/zap"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/internal/metadataproviders/kubeadm"
+	"github.com/open-telemetry/opentelemetry-collector-contrib/processor/resourcedetectionprocessor/internal"
+	"github.com/open-telemetry/opentelemetry-collector-contrib/processor/resourcedetectionprocessor/internal/kubeadm/internal/metadata"
+)
+
+const (
+	TypeStr                   = "kubeadm"
+	defaultConfigMapName      = "kubeadm-config"
+	defaultConfigMapNamespace = "kube-system"
```

Comment:
Nit: Do these defaults need to be in the resource detection processor? Could they instead be moved to [internal/metadataproviders/kubeadm/metadata.go](https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/35450/files#diff-2ae937bc91a72b68000cbcb79dacd82046881c6fa56beaee8f7a72095d657085)? It doesn't _appear_ to be necessary to pass these down to the `NewProvider` method, but maybe there's a scenario I'm not thinking of.

Reply:
The only scenario would be that we could make the configMap name and namespace configurable in the future. Currently that is not needed at all.

Technically we can move the default values to the internal package you mentioned, but I do not see much advantage technically there. Leaving it up to your decision.